src/platform/utils.py

```diff
@@ -0,0 +1,106 @@
+"""module to automatically wrap methods in autogenerated low-level code and re-expose them as high-level functions"""
+
+import inspect
+from urllib.parse import urljoin
+
+import do_sdk_platform
+from beartype import beartype
+from box import Box
+from deeporigin.auth import get_tokens
+from deeporigin.config import get_value
+from deeporigin.utils.core import _get_method
+
+
+@beartype
+def _get_api_client(*, api_name: str, configure: bool = True):
+    """return a configured client for the API we want to access
+
+    Args:
+        api_name (str): name of the API
+
+    Returns:
+        configured client
+    """
+
+    if configure:
+        configuration = do_sdk_platform.configuration.Configuration(
+            host=urljoin(get_value()["api_endpoint"], "/api"),
+            access_token=get_tokens()["access"],
+        )
+
+        client = do_sdk_platform.ApiClient(configuration=configuration)
+    else:
+        client = do_sdk_platform.ApiClient()
+
+    api_class = getattr(do_sdk_platform, api_name)
+    client = api_class(api_client=client)
+    return client
+
+
+@beartype
+def _get_client_methods(client) -> set:
+    """utility function to get methods from the client that return raw responses from the server"""
+    methods = set(
+        [
+            attr
+            for attr in dir(client)
+            if callable(getattr(client, attr))
+            and not attr.startswith("_")
+            and "without_preload_content" in attr
+        ]
+    )
```

Comment:
we're getting all (public) methods that have `without_prelload_content` in them so that we can use raw responses. typically errors in the spec mean that we can't reliably construct objects from them